pkg/datapath/linux/config/config.go

```diff
@@ -238,12 +238,16 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		if option.Config.EnableHostServicesUDP {
 			cDefinesMap["ENABLE_HOST_SERVICES_UDP"] = "1"
 		}
-		if option.Config.EnableHostServicesTCP && option.Config.EnableHostServicesUDP {
+		if option.Config.EnableHostServicesTCP && option.Config.EnableHostServicesUDP && !option.Config.BPFSocketLBHostnsOnly {
 			cDefinesMap["ENABLE_HOST_SERVICES_FULL"] = "1"
 		}
 		if option.Config.EnableHostServicesPeer {
 			cDefinesMap["ENABLE_HOST_SERVICES_PEER"] = "1"
 		}
+		if option.Config.BPFSocketLBHostnsOnly {
+			cDefinesMap["ENABLE_SOCKET_LB_HOST_ONLY"] = "1"
```

Comment:
Whichever new macro we introduce (see previous comment), we need to add it to the appropriate `MAX_XXX_OPTIONS` in `bpf/Makefile`.

Reply:
Hmm, not sure whether adding `ENABLE_SOCKET_LB_HOST_ONLY` macro in `MAX_LXC_OPTIONS` gives any value, as bpf_lxc for all permutations is already compiled with the per packet lb.

Reply:
Yep, I agree it's not mandatory. It would just have the benefit of being explicit (most people miss that `MAX_LXC_OPTIONS` already covers per-packet LB right now).

It's needed for `bpf_sock` however.